lib/gettext_i18n_rails/action_controller.rb

```diff
@@ -1,8 +1,23 @@
-class ActionController::Base
-  def set_gettext_locale
-    requested_locale = params[:locale] || session[:locale] || cookies[:locale] ||  request.env['HTTP_ACCEPT_LANGUAGE'] || I18n.default_locale
-    locale = FastGettext.set_locale(requested_locale)
-    session[:locale] = locale
-    I18n.locale = locale # some weird overwriting in action-controller makes this necessary ... see I18nProxy
+# Autoloading in initializers is deprecated on rails 6.0.  This delays initialization using the on_load
+# hooks, but does not change behaviour for existing rails versions.
+if defined?(Rails) && Rails::VERSION::MAJOR >= 6
+  ActiveSupport.on_load(:action_controller_base) do
```

Comment:
rails 5 does not have this ?

Reply:
It does, yes.  I figured we could just retain the existing loading functionality to reduce the chances of breakage on rails 5 apps.  I can change it to 5 if you'd prefer.